heare/developer/agent.py

```diff
@@ -230,9 +230,31 @@ def run(
 
         user_interface.handle_system_message(command_message)
 
+    # Initialize chat history, check if we're resuming a session
```

Comment:
we shouldn't be mucking with the context like this in the run loop. instead, let's move chat_history and tool_result_buffer into properties on AgentContext, and just accept the state as is. 

In that situation, resume is changing the contents of the agent context (or running on a different instance, we can figure that out).